tests/rc_string_ut/rc_string_ut.c

```diff
@@ -13,6 +15,11 @@
 #define ENABLE_MOCKS
 #include "c_pal/gballoc_hl.h"
 #include "c_pal/gballoc_hl_redirect.h"
+
+#include "umock_c/umock_c_prod.h"
```

Comment:
A wild umock_c_prod include has appeared!
Why do we need to do this here in the middle of the #define ENABLE_MOCKS
#undef ENABLE_MOCKS

?